src/conversion.rs

```diff
@@ -279,6 +280,7 @@ where
     T: PyClass + Clone,
 {
     fn extract(obj: &PyAny) -> PyResult<Self> {
+        #[allow(deprecated)]
         let cell: &PyCell<Self> = obj.downcast()?;
         Ok(unsafe { cell.try_borrow_unguarded()?.clone() })
```

Comment:
Looking at this, there's a potential soundness bug if for some horrendous reason `T::clone` user code ran arbitrary Python code which ended up mutating `obj`.

I wonder if we should just change this to do `.try_borrow()?.clone()` for now, at a slight efficiency loss. It would allow us to simplify here.

This method could also be updated to `extract_bound` at the same time!

Reply:
Yeah, I started porting that to `extract_bound` initially, but didn't see an equivalent to the `_unguarded` variant, so I left it. But your explination makes sense and fixing unsoundness is always good, so I changed that.